### ISSUE TITLE
topfile checker script crashes if you change non-topfiles in the topfile dir

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8563]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8563|
|Type|defect|
|Created|2016-07-05 15:19:40Z|
|Branch|https://github.com/twisted/twisted/tree/topfile-checker-fix-8563|

```
twisted/words/topfiles/README
----
Traceback (most recent call last):
  File "/buildslave/static-checkers/Twisted/bin/admin/check-topfile", line 22, in <module>
    CheckTopfileScript(print).main(sys.argv[1:])
  File "/buildslave/static-checkers/Twisted/twisted/python/_release.py", line 1111, in main
    if change.rsplit(".", 1)[1] in TOPFILE_TYPES:
IndexError: list index out of range
```

<details><summary>Searchable metadata</summary>

```
trac-id__8563 8563
type__defect defect
reporter__hawkowl hawkowl
priority__normal normal
milestone__totally_automated_release_infrastructure totally automated release infrastructure
branch__topfile_checker_fix_8563 topfile-checker-fix-8563
branch_author__ 
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__None None
time__1467731980401755 1467731980401755
changetime__1468189285369747 1468189285369747
version__None None
owner__hawkowl hawkowl
cc__radix
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

Looking good, please review.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set owner to @hawkowl|
|-|-|

Looks ok

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset e66797a57a988dbda23207b58bc31186bab2aeb5
```
#!CommitTicketReference repository="" revision="e66797a57a988dbda23207b58bc31186bab2aeb5"
Merge topfile-checker-fix-8563: Fix topfile checker.

Author: hawkowl
Reviewer: rodrigc
Fixes: #8563
```

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban commented|
|-|-|

Amber, I merged this on your behalf as I was annoyed by the topfile errors :)

Thanks for the fix.
